World.py

```diff
@@ -1,99 +1,101 @@
+        self.distribution: WorldDistribution = settings.distribution.world_dists[world_id]
 
         # rename a few attributes...
-        self.keysanity = settings.shuffle_smallkeys in ['keysanity', 'remove', 'any_dungeon', 'overworld', 'regional']
+        self.keysanity: bool = settings.shuffle_smallkeys in ['keysanity', 'remove', 'any_dungeon', 'overworld', 'regional']
         self.shuffle_silver_rupees = settings.shuffle_silver_rupees != 'vanilla'
-        self.check_beatable_only = settings.reachable_locations != 'all'
+        self.check_beatable_only: bool = settings.reachable_locations != 'all'
 
-        self.shuffle_special_interior_entrances = settings.shuffle_interior_entrances == 'all'
-        self.shuffle_interior_entrances = settings.shuffle_interior_entrances in ['simple', 'all']
+        self.shuffle_special_interior_entrances: bool = settings.shuffle_interior_entrances == 'all'
+        self.shuffle_interior_entrances: bool = settings.shuffle_interior_entrances in ['simple', 'all']
 
-        self.shuffle_special_dungeon_entrances = settings.shuffle_dungeon_entrances == 'all'
-        self.shuffle_dungeon_entrances = settings.shuffle_dungeon_entrances in ['simple', 'all']
+        self.shuffle_special_dungeon_entrances: bool = settings.shuffle_dungeon_entrances == 'all'
+        self.shuffle_dungeon_entrances: bool = settings.shuffle_dungeon_entrances in ['simple', 'all']
 
-        self.entrance_shuffle = (
+        self.entrance_shuffle: bool = bool(
             self.shuffle_interior_entrances or settings.shuffle_grotto_entrances or self.shuffle_dungeon_entrances
             or settings.shuffle_overworld_entrances or settings.shuffle_gerudo_valley_river_exit or settings.owl_drops or settings.warp_songs
             or settings.spawn_positions or (settings.shuffle_bosses != 'off')
         )
 
         self.mixed_pools_bosses = False # this setting is still in active development at https://github.com/Roman971/OoT-Randomizer
 
-        self.ensure_tod_access = self.shuffle_interior_entrances or settings.shuffle_overworld_entrances or settings.spawn_positions
-        self.disable_trade_revert = self.shuffle_interior_entrances or settings.shuffle_overworld_entrances or settings.adult_trade_shuffle
-        self.skip_child_zelda = 'Zeldas Letter' not in settings.shuffle_child_trade and \
-                                'Zeldas Letter' in self.distribution.starting_items
-        self.selected_adult_trade_item = ''
-        self.adult_trade_starting_inventory = ''
-
-        if (
-            settings.open_forest == 'closed'
-            and (
-                self.shuffle_special_interior_entrances or settings.shuffle_hideout_entrances or settings.shuffle_overworld_entrances
-                or settings.warp_songs or settings.spawn_positions
-            )
-        ):
+        self.ensure_tod_access: bool = bool(self.shuffle_interior_entrances or settings.shuffle_overworld_entrances or settings.spawn_positions)
```

Comment:
Why is specifying bool on the right side of the assignment done here?

Reply:
That's where that attribute is first defined and I tried to add a type hint for all class/instance attributes when they are defined. `World` just has a sprawling `__init__` so it looks a bit weirder.

Reply:
Also, I had to wrap the thing in a `bool()` call so that the result of it was actually a bool since two of those settings it uses are actually a list and calling `or` just gets the first result that would be considered True, it doesn't itself become a bool.

Reply:
Should the same thing not then be done for the ones below this like `disable_trade_revert`?

Reply:
It probably should be done for `disable_trade_revert` for the same reason. I'm not sure why I only did it for `ensure_tod_access`, I think my IDE just happened to bit about that one at some point?

Reply:
Just looked at this harder and `disable_trade_revert` is an or of 3 booleans which is why it didn't need the `bool()` call. `ensure_tod_access` is an or of 2 booleans and a list, so it does.